build/Microsoft.DotNet.Cli.Prepare.targets

```diff
@@ -1,47 +1,98 @@
     </PropertyGroup>
 
-    <!-- Generates BuildVersion Item
-        BuildVersion Metadata:
-          - Major
-          - Minor
-          - Patch
-          - ReleaseSuffix
-          - CommitHash
-          - CommitCount
-          - VersionSuffix
-          - SimpleVersion
-          - NugetVersion
-          - MsiVersion            -->
     <GenerateBuildVersionInfo RepoRoot="$(RepoRoot)">
-        <Output TaskParameter="OutputBuildVersionInfo"
-                ItemName="BuildVersion" />
+      <Output TaskParameter="VersionMajor" PropertyName="VersionMajor" />
+      <Output TaskParameter="VersionMinor" PropertyName="VersionMinor" />
+      <Output TaskParameter="VersionPatch" PropertyName="VersionPatch" />
+      <Output TaskParameter="CommitHash" PropertyName="CommitHash" />
+      <Output TaskParameter="CommitCount" PropertyName="CommitCount" />
+      <Output TaskParameter="ReleaseSuffix" PropertyName="ReleaseSuffix" />
+      <Output TaskParameter="VersionSuffix" PropertyName="VersionSuffix" />
+      <Output TaskParameter="SimpleVersion" PropertyName="SimpleVersion" />
+      <Output TaskParameter="NugetVersion" PropertyName="NugetVersion" />
+      <Output TaskParameter="MsiVersion" PropertyName="MsiVersion" />
+      <Output TaskParameter="VersionBadgeMoniker" PropertyName="VersionBadgeMoniker" />
+      <Output TaskParameter="Channel" PropertyName="Channel" />
+      <Output TaskParameter="BranchName" PropertyName="BranchName" />
     </GenerateBuildVersionInfo>
 
     <PropertyGroup>
-      <SdkVersion>@(BuildVersion ->'%(NugetVersion)')</SdkVersion>
+      <VersionBadge>$(BaseOutputDirectory)/$(VersionBadgeMoniker)_$(Configuration)_version_badge.svg</VersionBadge>
+      <SdkVersion>$(NugetVersion)</SdkVersion>
+
+      <ArchiveExtension Condition=" '$(OSName)' == 'win' ">.zip</ArchiveExtension>
+      <ArchiveExtension Condition=" '$(OSName)' != 'win' ">.tar.gz</ArchiveExtension>
+
+      <InstallerExtension Condition=" '$(OSName)' == 'win' ">.msi</InstallerExtension>
+      <InstallerExtension Condition=" '$(OSName)' == 'osx' ">.pkg</InstallerExtension>
+      <InstallerExtension Condition=" '$(OSName)' == 'ubuntu' ">.deb</InstallerExtension>
+
+      <BundleExtension Condition=" '$(OSName)' == 'win' ">.exe</BundleExtension>
+      <BundleExtension Condition=" '$(OSName)' == 'osx' ">$(InstallerExtension)</BundleExtension>
+      <BundleExtension Condition=" '$(OSName)' == 'ubuntu' ">$(InstallerExtension)</BundleExtension>
+
+      <ProductMonikerRid Condition=" '$(Rid)' == 'ubuntu.16.04-x64' Or '$(Rid)' == 'fedora.23-x64' Or '$(Rid)' == 'opensuse.13.2-x64' ">$(Rid)</ProductMonikerRid>
+      <ProductMonikerRid Condition=" '$(ProductMonikerRid)' == '' ">$(OSName)-$(Architecture)</ProductMonikerRid>
+
+      <!-- Generated Installers + Archives -->
+      <SdkCompressedFile>$(PackagesDirectory)/dotnet-sdk-$(ProductMonikerRid).$(SdkVersion)$(ArchiveExtension)</SdkCompressedFile>
```

Comment:
While I didn't double check all the values here this doesn't look too unreasonable. Yes it might be a little more verbose but I think this is likely the best translation you could expect. It maybe possible to reduce a little more of the redundancy but I didn't dig too deeply.